application/apps/rustcore/ts-bindings/spec/session.benchmark.spec.ts

```diff
@@ -0,0 +1,306 @@
+                                if (startupStream instanceof Error) throw startupStream;
+
+                                const startupEvents = session.getEvents();
+                                if (startupEvents instanceof Error) throw startupEvents;
+                                startupStream.observe(
+                                    new Factory.Stream()
+                                        .asText()
+                                        .process({
+                                            command: `less ${tmpobj1.name}`,
+                                            cwd: process.cwd(),
+                                            envs: process.env as { [key: string]: string },
+                                        })
+                                        .get()
+                                        .sterilized()
+                                );
+                                const startupResults = measurement();
+                                const startupReport = performanceReport(testName, startupResults.ms, test.expectation_ms, `${home_dir}/${test.file}`);
+                                finish(startupSession, done, startupReport ? undefined : new Error(`${testName} is fail`));
+                                break;
+                            case 'Stream - shutdown measurement':
+                                const tmpobj2 = createSampleFile(
+                                    5000,
+                                    logger,
+                                    (i: number) => `some line data: ${i}\n`
+                                );
+
+                                const shutdownSession = await Session.create();
+                                shutdownSession.debug(true, testName);
+
+                                const shutdownStream = shutdownSession.getStream();
+                                if (shutdownStream instanceof Error) throw shutdownStream;
+
+                                shutdownStream.observe(
+                                    new Factory.Stream()
+                                        .asText()
+                                        .process({
+                                            command: `less ${tmpobj2.name}`,
+                                            cwd: process.cwd(),
+                                            envs: process.env as { [key: string]: string },
+                                        })
+                                        .get()
+                                        .sterilized()
+                                );
+                                const shutdownResults = measurement();
+                                const shutdownReport = performanceReport(testName, shutdownResults.ms, test.expectation_ms, `${home_dir}/${test.file}`);
+                                finish(shutdownSession, done, shutdownReport ? undefined : new Error(`${testName} is fail`));
+                                break;
```

Comment:
you are leaving a test without closing `session` which was created above at line 42